app/questionnaire/placeholder_parser.py

```diff
@@ -166,32 +171,70 @@ def _parse_transforms(
 
         return transformed_value
 
+    @staticmethod
     def _resolve_value_source_list(
-        self, value_source_list: list[dict]
+        value_source_list: list[dict], value_source_resolver: ValueSourceResolver
     ) -> list[ValueSourceTypes]:
         values: list[ValueSourceTypes] = []
         for value_source in value_source_list:
-            value = self._value_source_resolver.resolve(value_source)
+            value = value_source_resolver.resolve(value_source)
             if isinstance(value, list):
                 values.extend(value)
             else:
                 values.append(value)
         return values
 
     def _get_routing_path_block_ids(
-        self, data: Sequence[Mapping], sections_to_ignore: list | None = None
+        self,
+        data: Sequence[Mapping],
+        sections_to_ignore: list | None = None,
     ) -> dict[tuple, tuple[str, ...]] | None:
+        if not self._location:
+            return {}
+
+        dependent_sections = (
+            self._schema.calculated_summary_section_dependencies_by_block[
+                self._location.section_id
+            ]
+        )
+
+        return get_block_ids_for_dependencies(
+            location=self._location,
+            progress_store=self._progress_store,
+            sections_to_ignore=sections_to_ignore,
+            data=data,
+            path_finder=self._path_finder,
+            source_type="calculated_summary",
+            ignore_keys=None,
+            dependent_sections=dependent_sections,
+        )
+
+    def _all_value_sources_metadata(self, placeholder: Mapping) -> bool:
+        sources = self._schema.get_values_for_key(placeholder, key="source")
+        return all(source == "metadata" for source in sources)
+
+    def _get_value_source_resolver_for_transform(
+        self, transform: Mapping
+    ) -> ValueSourceResolver:
         if self._location:
-            return get_block_ids_for_calculated_summary_dependencies(
-                schema=self._schema,
+            dependent_sections = self._schema.placeholder_section_dependencies_by_block[
+                self._location.section_id
+            ]
+            block_ids = get_block_ids_for_dependencies(
                 location=self._location,
                 progress_store=self._progress_store,
-                sections_to_ignore=sections_to_ignore,
-                data=data,
                 path_finder=self._path_finder,
+                data=transform,
+                source_type="answers",
+                dependent_sections=dependent_sections,
+            )
+            self._routing_path_block_ids_by_section_key.update(block_ids)
+            routing_path_block_ids: list = [
+                value for values in block_ids.values() for value in values
+            ]
+            return self._get_value_source_resolver(
+                routing_path_block_ids=OrderedSet(routing_path_block_ids),
```

Comment:
Removed Ordered set call and now it should use the routing block ids form the main VSR. Not too sure if I did it right

Reply:
Updated: OrderedSet is used again because VSR routing path block ids is an Iterable[STR] and routing path ids in get_value_source_resolver is typed as an Ordered Set

Reply:
That type should just be `Iterable[str]`

Reply:
Updated 👍